File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& str) {
        map<string,vector<string>>m;
        for(int i=0;i<str.size();i++)
        {
            string s=str[i];
            sort(str[i].begin(),str[i].end());
           m[str[i]].push_back(s);   
        }
        vector<vector<string>>res;
        for(auto a:m)
        {
            vector<string>temp;
            for(auto b:a.second)
            {
                temp.push_back(b);
            }
            res.push_back(temp);
            
        }
        
        
        return res;
        
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>>mp=new HashMap<>();
        for(int i=0;i<strs.length;i++){
            char[] str=strs[i].toCharArray();
            Arrays.sort(str);
            String newString=new String(str);
            List<String> temp=mp.getOrDefault(newString,new ArrayList<String>());
            temp.add(strs[i]);
            mp.put(newString,temp);
        }
        List<List<String>> ans=new ArrayList<>();
        for(Map.Entry<String,List<String>> mpElement: mp.entrySet()){
            ans.add(mpElement.getValue());
            
        }

        return ans;
    }
}